packaging/markers.py

```diff
@@ -219,6 +220,16 @@ def _get_env(environment: Dict[str, str], name: str) -> str:
     return value
 
 
+def _normalize(*values: str, key: str) -> Tuple[str, ...]:
+    # PEP 685 – Comparison of extra names for optional distribution dependencies
+    # https://peps.python.org/pep-0685/
+    if key == "extra":
+        return tuple(re.sub(r"[-_.]+", "-", v).lower() for v in values)
```

Comment:
I don't think we need to re-implement the normalization when `.utils.canonicalize_name()` does the work already.

```suggestion
        return tuple(map(canonicalize_name, values))
```

Reply:
Awesome. I still prefer a comprehension over map if that's fine with you.

Reply:
Added a fixup commit.